decoding_suffix.py

```diff
@@ -241,12 +250,16 @@ def decode(model, tokenizer, device, x="", z="", constraints=None, args=None, sy
                 model, y_logits_, args.topk, soft_forward_x, x_model_past, tokenizer, extra_mask=None, bad_mask=None)
             text_post = text
             for bi in range(args.batch_size):
-                prompt = x + " " + text_post[bi]
+                prompt = sys_prompt + x + " " + text_post[bi] + user_prompt_suffix
```

Comment:
Ah, another comment here I meant to say. I noticed this code before did not have any system prompt at all. This seems wrong as a few lines below this is tokenized into `input_ids` and passed into `model.generate` from scratch, so my correction here adds both the system prompt and what I'm calling the user prompt suffix  `[/INST]`. I also wonder if the addition of the space here (`+ " " +`) could be causing some of the transferability issues? I quickly tried removing it but it didn't seem to change much

Reply:
Instead of `+ " " +` here, I think the most correct way would be to take concatenated `[prefix, hard-projected adversarial suffix, and user prompt suffix]` and decode it with the tokenizer.

Also these replacements in [get_text_from_logits](https://github.com/Yu-Fangxu/COLD-Attack/blob/63ba3cfab91d3f9f281c28b5c83e60a024d0f866/util.py#L138C1-L139C23) look a little strange. This is called before `decode_with_model_topk` returns the text a few lines above this part (and in other places):

```
        text_i = text_i.replace('\n', ' ')
        text_i += ". "
```